packages/core/tests/types/types.test.ts

```diff
@@ -290,3 +290,40 @@ const testRule: RuleDefinition<{
 };
 
 testRule.meta satisfies RulesMeta | undefined;
+
+const testRuleWithInvalidDefaultOptions: RuleDefinition<{
+	LangOptions: TestLanguageOptions;
+	Code: TestSourceCode;
+	RuleOptions: [{ foo: string; bar: number }];
+	Visitor: TestRuleVisitor;
+	Node: TestNode;
+	MessageIds: "badFoo" | "wrongBar";
+	ExtRuleDocs: never;
+}> = {
+	meta: {
+		type: "problem",
+		schema: [
+			{
+				type: "object",
+				properties: {
+					foo: {
+						type: "string",
+					},
+					bar: {
+						type: "integer",
+					},
+				},
+				additionalProperties: false,
+			},
+		],
+
+		// @ts-expect-error invalid default option "baz"
+		defaultOptions: [{ foo: "always", bar: 5, baz: "invalid" }],
```

Comment:
Maybe we can format it like this so the test will check that the types of `foo` and `bar` are correct, and only `baz` is invalid.

```suggestion
		defaultOptions: [
			{
				foo: "always",
				bar: 5,
				// @ts-expect-error invalid default option "baz"
				baz: "invalid",
			},
		],
```